pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/MailboxService.java

```diff
@@ -159,4 +175,26 @@ public ReceivingMailbox getReceivingMailbox(String mailboxId) {
   public void releaseReceivingMailbox(ReceivingMailbox mailbox) {
     _receivingMailboxCache.invalidate(mailbox.getId());
   }
+
+  private static Duration getPingerPeriod(PinotConfiguration config) {
+    long pingerPeriodSecs = config.getProperty(
+        CommonConstants.MultiStageQueryRunner.KEY_OF_PINGER_PERIOD_SECONDS,
+        CommonConstants.MultiStageQueryRunner.DEFAULT_PINGER_PERIOD_SECONDS);
+    return Duration.ofSeconds(pingerPeriodSecs);
+  }
+
+  private static Duration getIdleTimeout(PinotConfiguration config) {
+    long channelIdleTimeoutSeconds = config.getProperty(
+        CommonConstants.MultiStageQueryRunner.KEY_OF_CHANNEL_IDLE_TIMEOUT_SECONDS,
+        CommonConstants.MultiStageQueryRunner.DEFAULT_CHANNEL_IDLE_TIMEOUT_SECONDS);
+    if (channelIdleTimeoutSeconds > 0) {
+      return Duration.ofSeconds(channelIdleTimeoutSeconds);
+    }
+    Duration pingerPeriod = getPingerPeriod(config);
+    if (pingerPeriod.isNegative() || pingerPeriod.isZero()) {
+      return Duration.ofMinutes(30);
+    } else {
+      return pingerPeriod.multipliedBy(2);
+    }
```

Comment:
Is this because we always want to override the idle timeout and never use the default gRPC value?

Reply:
Okay, that makes sense - I think this is fine.